src/Controller/Rest/ArticleController.php

```diff
@@ -0,0 +1,115 @@
+<?php
+/**
+ * Created by IntelliJ IDEA.
+ * User: N-138
+ * Date: 05/07/2021
+ * Time: 4:37 PM
+ */
+
+namespace App\Controller\Rest;
+
+use App\Entity\Article;
+use App\Form\ArticleType;
+use FOS\RestBundle\Controller\AbstractFOSRestController;
+use FOS\RestBundle\Controller\Annotations as Rest;
+use FOS\RestBundle\View\View;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\ParamConverter;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+
+class ArticleController extends AbstractFOSRestController
+{
+    /**
+     * @Rest\Get("/articles")
+     * @param Request $request
+     * @return View
+     */
+    public function getArticles(Request $request)
+    {
+        $title = $request->get('title');
+        $entityManager = $this->getDoctrine()->getManager();
+
+        if (!$title) {
+            $articles = $entityManager->getRepository(Article::class)->findAll();
+        } else {
+            $articles = $this->getDoctrine()
+                ->getManager()
+                ->getRepository(Article::class)
+                ->searchByTitle($title);
+        }
+
+        return View::create($articles, Response::HTTP_OK);
+    }
+
+    /**
+     * @Rest\Get("/articles/{id}")
+     * @ParamConverter("article")
+     * @param Article $article
+     * @return View
+     */
+    public function getArticleById(Article $article)
+    {
+        return View::create($article, Response::HTTP_OK);
+    }
+
+    /**
+     * @Rest\Post("/articles")
+     * @param Request $request
+     * @return View
+     */
+    public function postArticle(Request $request)
+    {
+        $body = $request->getContent();
+        $data = json_decode($body, true);
+
+//        var_dump($data);
+//        var_dump($request->request->all());
+
+        $form = $this->createForm(ArticleType::class, new Article());
+        $form->submit($data);
+
+        if (!$form->isValid()) {
+            return $this->view('Article is invalid', Response::HTTP_NOT_FOUND);
```

Comment:
sorry @robredido i think 404 response status does not fit for this error. shouldn't it be 400(HTTP_BAD_REQUEST)?